benchmarks/cpp/tensorexpr/bench_gemm.cpp

```diff
@@ -127,6 +127,216 @@ BENCHMARK_DEFINE_F(Gemm, TensorExprTile32x32)(benchmark::State& state) {
   }
 }
 
+BENCHMARK_DEFINE_F(Gemm, TensorExprTile4x16)(benchmark::State& state) {
+  te::KernelScope ks;
+
+  te::Placeholder AP(te::BufHandle("A", {M, K}, te::kFloat));
+  te::Placeholder BP(te::BufHandle("B", {K, N}, te::kFloat));
+  te::Tensor* CT = te::Reduce(
+      "gemm",
+      {{M, "M"}, {N, "N"}},
+      te::Sum(),
+      [&](const te::ExprHandle& m,
+          const te::ExprHandle& n,
+          const te::ExprHandle& k) { return AP.load(m, k) * BP.load(k, n); },
+      {{K, "K"}});
+  te::LoopNest loop({CT});
+
+  {
+    auto const& loops = loop.getLoopStmtsFor(CT);
+    te::For* m = loops[0];
+    te::For* mo;
+    te::For* mi;
+    loop.splitWithMask(m, 4, &mo, &mi);
+  }
+  {
+    auto const& loops = loop.getLoopStmtsFor(CT);
+    te::For* n = loops[2];
+    te::For* no;
+    te::For* ni;
+    loop.splitWithMask(n, 16, &no, &ni);
+  }
+  // mo, mi, no, ni, k ->
+  // mo, no, mi, ni, k
+  {
+    auto const& loops = loop.getLoopStmtsFor(CT);
+    te::For* mi = loops[1];
+    te::For* no = loops[2];
+    loop.reorderAxis(mi, no);
+  }
+  // mo, no, mi, ni, k ->
+  // mo, no, mi, k, ni
+  {
+    auto const& loops = loop.getLoopStmtsFor(CT);
+    te::For* ni = loops[3];
+    te::For* k = loops[4];
+    loop.reorderAxis(ni, k);
+  }
+  // mo, no, mi, k, ni ->
+  // mo, no, k, mi, ni
+  {
+    auto const& loops = loop.getLoopStmtsFor(CT);
+    te::For* mi = loops[2];
+    te::For* k = loops[3];
+    loop.reorderAxis(mi, k);
+  }
```

Comment:
This code to tile in a specific way, 4x16 here, and reorder the loops appropriately, is being used repeatedly. How about creating a utility to tile the loops given the tile sizes?

Reply:
I could factor something out here I guess, but I'd like us to improve the library API.  In tvm this whole giant pile of stuff is three lines:
```
mo, mi = s[C].split(m, 4)
no, ni = s[C].split(n, 16)
s[C].reorder(mo, no, k, mi, ni)
```